mlir/include/mlir/Dialect/GPU/IR/CompilationInterfaces.h

```diff
@@ -100,6 +111,10 @@ class TargetOptions {
   /// Compilation process target representation.
   CompilationTarget compilationTarget;
 
+  /// Callback for obtaining the parent symbol table of all the GPU modules
+  /// being serialized.
+  function_ref<SymbolTable *()> getSymbolTableCallback;
```

Comment:
Is this safe?
function_ref says:
/// An efficient, type-erasing, non-owning reference to a callable. This is
/// intended for use as the type of a function parameter that is not used
/// after the function in question returns.
///
/// This class does not own the callable, so it is not in general safe to store
/// a function_ref.
So who owns  `= {}` here?

Reply:
I'm seeing clang-tidy warnings/crashes in similar usage - https://github.com/llvm/llvm-project/issues/71363

Reply:
It can be changed. However, this class is only meant to be used as a temp to pass arguments with the caller being in charge of making sure the object remains alive and not really to store data.